ui-lib/src/main/java/cash/z/ecc/ui/MainActivity.kt

```diff
@@ -4,23 +4,37 @@ import android.os.Bundle
 import androidx.activity.ComponentActivity
 import androidx.activity.compose.setContent
 import androidx.activity.viewModels
+import androidx.compose.runtime.collectAsState
+import cash.z.ecc.ui.screen.home.view.Home
+import cash.z.ecc.ui.screen.home.viewmodel.WalletViewModel
 import cash.z.ecc.ui.screen.onboarding.view.Onboarding
 import cash.z.ecc.ui.screen.onboarding.viewmodel.OnboardingViewModel
 import cash.z.ecc.ui.theme.ZcashTheme
 
 class MainActivity : ComponentActivity() {
 
+    private val walletViewModel by viewModels<WalletViewModel>()
+
     private val onboardingViewModel by viewModels<OnboardingViewModel>()
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContent {
             ZcashTheme {
-                Onboarding(
-                    onboardingState = onboardingViewModel.onboardingState,
-                    onImportWallet = { TODO("Implement wallet import") },
-                    onCreateWallet = { TODO("Implement wallet create") }
-                )
+                if (null == walletViewModel.persistableWallet.collectAsState(null).value) {
+                    // Optimized path to get to onboarding as quickly as possible
+                    Onboarding(
+                        onboardingState = onboardingViewModel.onboardingState,
+                        onImportWallet = { TODO("Implement wallet import") },
+                        onCreateWallet = { TODO("Implement wallet create") }
+                    )
+                } else {
+                    if (null == walletViewModel.synchronizer.collectAsState(null).value) {
+                        // Continue displaying splash screen
```

Comment:
Something we found useful is to have a "NoScreen" which also captures a whole can of worms of initialization errors like db migration problems, secure storage not being decrypted in a timely fashion (that's an iOS problem btw), and other weird stuff that will actually prevent the user to use the app. 

This NoScreen allows you to help the user recover from tech disaster and access the recovery phrase. Would this place be a good one to introduce such a screen?

Reply:
Good idea, yes this could be a place to introduce something like that 👍